presto-main/src/main/java/com/facebook/presto/sql/planner/plan/TableScanNode.java

```diff
@@ -82,9 +85,10 @@ public TableScanNode(
             List<Symbol> outputs,
             Map<Symbol, ColumnHandle> assignments,
             TupleDomain<ColumnHandle> currentConstraint,
-            TupleDomain<ColumnHandle> enforcedConstraint)
+            TupleDomain<ColumnHandle> enforcedConstraint,
+            Map<Symbol, List<SubfieldPath>> requiredSubfieldPaths)
```

Comment:
Let's merge this constructor with the one that also takes temporaryTable argument.